src/test/java/org/motechproject/prevac/osgi/PrevacLifecycleListenerIT.java

```diff
@@ -0,0 +1,101 @@
+import org.junit.runner.RunWith;
+import org.motechproject.prevac.domain.Clinic;
+import org.motechproject.prevac.domain.Subject;
+import org.motechproject.prevac.domain.Visit;
+import org.motechproject.prevac.domain.enums.Language;
+import org.motechproject.prevac.domain.enums.VisitType;
+import org.motechproject.prevac.repository.ClinicDataService;
+import org.motechproject.prevac.repository.UnscheduledVisitDataService;
+import org.motechproject.prevac.repository.VisitDataService;
+import org.motechproject.prevac.utils.SubjectUtil;
+import org.motechproject.testing.osgi.BasePaxIT;
+import org.motechproject.testing.osgi.container.MotechNativeTestContainerFactory;
+import org.ops4j.pax.exam.ExamFactory;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+
+import javax.inject.Inject;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.motechproject.prevac.utils.ClinicUtil.createClinic;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+@ExamFactory(MotechNativeTestContainerFactory.class)
+public class PrevacLifecycleListenerIT extends BasePaxIT {
+
+    @Inject
+    private ClinicDataService clinicDataService;
+
+    @Inject
+    private VisitDataService visitDataService;
+
+    @Inject
+    private UnscheduledVisitDataService unscheduledVisitDataService;
+
+    @Before
+    public void cleanBefore() {
+        clearDataBase();
+    }
+
+    @After
+    public void cleanAfter() {
+        clearDataBase();
+    }
+
+    @Test
+    public void shouldUpdateclinicInVisits() {
+        List<Clinic> clinicList = clinicDataService.retrieveAll();
+        List<Visit> visitList = visitDataService.retrieveAll();
+        assertTrue(clinicList.isEmpty() && visitList.isEmpty());
+
+        String siteId = "siteItToFind";
+
+        Subject subject = SubjectUtil.createSubject("1000000161", "Michal", "729402018364", Language.English);
+        subject.setSiteId(siteId);
+        Subject subjectNotInCreatedClinic = SubjectUtil.createSubject("1000000162", "Maciek", "129402018364", Language.English);
+
+        createVisit(subject, VisitType.SCREENING, new LocalDate(2014, 10, 17), new LocalDate(2014, 10, 18));
+        createVisit(subject, VisitType.PRIME_VACCINATION_DAY, new LocalDate(2014, 10 , 19), new LocalDate(2014, 10, 20));
+        createVisit(subject, VisitType.PRIME_VACCINATION_FIRST_FOLLOW_UP_VISIT, new LocalDate(2014, 10, 19), new LocalDate(2014, 10, 20));
+        createVisit(subjectNotInCreatedClinic, VisitType.PRIME_VACCINATION_DAY, new LocalDate(2014, 10, 20), new LocalDate(2014, 10, 21));
+
+        Clinic clinic = createClinic(siteId, "location");
+        clinic = clinicDataService.create(clinic);
+
+        visitList = visitDataService.retrieveAll();
+
+        for (Visit visit : visitList) {
+            if (visit.getSubject() == subject) {
```

Comment:
you should not use == when comparing objects, use equals instead, you can compare only subjectId here